lib/bank_account/money_market.rb

```diff
@@ -0,0 +1,87 @@
+module Bank
+  # inherits from SavingsAccount which inherits from Account
+  class MoneyMarketAccount < SavingsAccount
+    attr_accessor :transaction_count, :day_of_month
+
+    # inherits from Account class
+    def initialize(id, initial_balance)
+      super
+      raise ArgumentError.new "INSUFFICIENT FUNDS" unless initial_balance >= 10000
+      # additional variables specific to this class
+      @transaction_count = 0
+      @day_of_month = 1
+    end
+
+    # checks for low balace before allowing withdrawal
+    # adds to time and transaction counts
+    def withdraw(amount)
+      time_passes
+      low_balance
+      transaction
+      if @balance < amount
+        puts "INSUFFICIENT FUNDS"
+        return @balance
+      else
+        @balance -= amount
+        if @balance < 10000
+          @balance -= 100
+          low_balance
+        else
+          return @balance
+        end
+      end
+    end
+
+    # adds amount to balance
+    def deposit(amount)
+      time_passes
+      # allows deposit transaction for balance below lower limit
+      # without adding to count
+      if @balance < 10000
+        return @balance += amount
+      # adds to transaction if balance is above lower limit
+      else
+        transaction
+        return @balance += amount
+      end
```

Comment:
It looks like you write `return @balance += amount` twice. You might want to try rewriting the conditional so you only need it once.